x-pack/solutions/observability/plugins/streams_app/public/components/stream_detail_enrichment/processors/grok/grok_ai_suggestions.tsx

```diff
@@ -0,0 +1,322 @@
+            {
+              defaultMessage: '✨ Generate patterns',
+            }
+          )}
+        </EuiButton>
+      </EuiFlexItem>
+      {connectors && connectors.length > 1 && (
+        <EuiFlexItem grow={false}>
+          <EuiPopover
+            id={splitButtonPopoverId}
+            isOpen={isPopoverOpen}
+            button={
+              <EuiButtonIcon
+                data-test-subj="streamsAppGrokAiPickConnectorButton"
+                onClick={onButtonClick}
+                display="base"
+                size="s"
+                iconType="boxesVertical"
+                aria-label={i18n.translate('xpack.streams.refreshButton.euiButtonIcon.moreLabel', {
+                  defaultMessage: 'More',
+                })}
+              />
+            }
+          >
+            <EuiContextMenuPanel
+              size="s"
+              items={connectors.map((connector) => (
+                <EuiContextMenuItem
+                  key={connector.id}
+                  icon={connector.id === currentConnector ? 'check' : 'empty'}
+                  onClick={() => {
+                    selectConnector?.(connector.id);
+                    closePopover();
+                  }}
+                >
+                  {connector.name}
+                </EuiContextMenuItem>
+              ))}
+            />
+          </EuiPopover>
+        </EuiFlexItem>
+      )}
+    </EuiFlexGroup>
+  );
+};
+
+export function useAiEnabled() {
+  const { dependencies } = useKibana();
+  const { observabilityAIAssistant } = dependencies.start;
+
+  const aiAssistantEnabled = observabilityAIAssistant?.service.isEnabled();
+
+  const genAiConnectors = observabilityAIAssistant?.useGenAIConnectors();
+
+  return aiAssistantEnabled && (genAiConnectors?.connectors || []).length > 0;
+}
+
+export function GrokAiSuggestions({
+  definition,
+  refreshSimulation,
+  samples,
+  extraButtons,
+}: {
+  definition: StreamDefinition;
+  refreshSimulation: UseProcessingSimulatorReturn['refreshSimulation'];
+  samples: RecursiveRecord[];
+  extraButtons?: React.ReactNode;
+}) {
+  const { dependencies } = useKibana();
+  const {
+    streams: { streamsRepositoryClient },
+    observabilityAIAssistant,
+  } = dependencies.start;
+
+  const fieldValue = useWatch<ProcessorFormState, 'field'>({ name: 'field' });
+  const form = useFormContext<GrokFormState>();
+
+  const genAiConnectors = observabilityAIAssistant?.useGenAIConnectors();
+  const currentConnector = genAiConnectors?.selectedConnector;
+
+  const [isLoadingSuggestions, setSuggestionsLoading] = useState(false);
+  const [suggestionsError, setSuggestionsError] = useState<Error | undefined>();
+  const [suggestions, setSuggestions] = useState<
+    { patterns: string[]; simulations: any[] } | undefined
+  >();
+  const [blocklist, setBlocklist] = useState<Set<string>>(new Set());
+
+  const abortController = useAbortController();
+
+  const refreshSuggestions = useCallback(() => {
```

Comment:
is it not possible to use `useStreamsAppFetch()` here?

Reply:
`useStreamsAppFetch` triggers right away, but this should be on-demand only, which I didn't find a nice way to achieve on the consumer side. I pondered adding this functionality to `useAbortableAsync` (some kind of `fetchOnMount` option), but it felt like overloading it too much. Maybe I missed an easy way to achieve this, happy to improve.

Reply:
I guess it would be something like `const [ refreshId, setRefreshId ] = useState(0);`

and then:

```ts

setRefreshId(prev => prev+1);

...

if (refreshId === 0) {
	return;
}
```

Not very pretty but beats re-implementing all the state management I guess? Is there no other state to use to determine when to fetch? (e.g. if the form values have changed from the original form values)

Reply:
Hmm, but I never want the suggestions to be triggered without the user clicking the button. It seems like the wrong tool for the job - not sure how to effectively abstract it out, maybe we should create a `useStreamsFetchCallback` which returns `{ value, isLoading, error, fetch }` or so. Wdyt?

Reply:
yeah, that's probably better, but fine to postpone that as well. that probably could be captured in `useAbortableAsyncCallback`

Reply:
You could have probably used the `useAsyncFn` hook here, as it abstract all the part for error handling and loading state and gives you the explicit trigger to send the request. But I wouldn't mind if this doesn't update now, with the state management refactor this will be handled at the processor level to prevent impossible side effects.